setup.py

```diff
@@ -85,7 +85,7 @@ def long_description():
         "Issues": "https://github.com/zulip/zulip-terminal/issues",
         "Hot Keys": "https://github.com/zulip/zulip-terminal/blob/main/docs/hotkeys.md",
     },
-    python_requires=">=3.7, <3.12",
+    python_requires=">=3.7",
```

Comment:
This seems to be an agreed-upon approach; the challenges are either keeping up to date with issues from newer python versions (and pinning until tested), or relying upon users to report issues with those newer versions.

In the past we staged support for more recent versions in CI, before making a release. With this change it'd be useful to document that certain versions are supported, but to report success and details with other pythons - perhaps similar to how it's great to hear reports from other terminal emulators that work well.